src/core/workspace.c

```diff
@@ -1065,39 +1063,15 @@ focus_ancestor_or_mru_window (MetaWorkspace *workspace,
           meta_window_focus (ancestor, timestamp);
 
           /* Also raise the window if in click-to-focus */
-          if (meta_prefs_get_focus_mode () == META_FOCUS_MODE_CLICK)
+          if (meta_prefs_get_focus_mode () == G_DESKTOP_FOCUS_MODE_CLICK)
             meta_window_raise (ancestor);
 
           return;
         }
     }
 
-  /* No ancestor, look for the MRU window */
-  tmp = workspace->mru_list;
-
-  while (tmp)
-    {
-      MetaWindow* tmp_window;
-      tmp_window = ((MetaWindow*) tmp->data);
-      if (tmp_window != not_this_one           &&
-          meta_window_showing_on_its_workspace (tmp_window) &&
-          tmp_window->type != META_WINDOW_DOCK &&
-          tmp_window->type != META_WINDOW_DESKTOP)
-        {
-          window = tmp->data;
-          break;
-        }
-      else if (tmp_window != not_this_one      &&
-               desktop_window == NULL          &&
-               meta_window_showing_on_its_workspace (tmp_window) &&
-               tmp_window->type == META_WINDOW_DESKTOP)
-        {
-          /* Found the most recently used desktop window */
-          desktop_window = tmp_window;
-        }
-
-      tmp = tmp->next;
-    }
+  window = meta_stack_get_default_focus_window (workspace->screen->stack,
+                                                workspace, NULL);
 
   /* If no window was found, default to the MRU desktop-window */
   if (window == NULL)
```

Comment:
This part was removed in the original commit, and needs to be removed here too, as there's no longer a `desktop_window`:
```
core/workspace.c:1078:14: error: ‘desktop_window’ undeclared (first use in this function)
 1078 |     window = desktop_window;
      |              ^~~~~~~~~~~~~~
```